luxtronik/__init__.py

```diff
@@ -38,24 +39,40 @@ def is_socket_closed(sock: socket.socket) -> bool:
     """Check is socket closed."""
     try:
         # this will try to read bytes without blocking and also without removing them from buffer
-        data = sock.recv(
-            LUXTRONIK_SOCKET_READ_SIZE_PEEK, socket.MSG_DONTWAIT | socket.MSG_PEEK
-        )
+        data = sock.recv(LUXTRONIK_SOCKET_READ_SIZE_PEEK, socket.MSG_DONTWAIT | socket.MSG_PEEK)
         if len(data) == 0:
             return True
     except BlockingIOError:
         return False  # socket is open and reading from it would block
     except ConnectionResetError:  # pylint: disable=broad-except
         return True  # socket was closed for some other reason
     except Exception as err:  # pylint: disable=broad-except
-        LOGGER.exception(
-            "Unexpected exception when checking if socket is closed", exc_info=err
-        )
+        LOGGER.exception("Unexpected exception when checking if socket is closed", exc_info=err)
         return False
     return False
 
 
-class Luxtronik:
+class LuxtronikData:
+    """
+    Collection of parameters, calculations and visiblities.
+    Also provide some high level access functions to their data values.
+    """
+
+    def __init__(self, parameters, calculations, visibilities):
+        self.parameters = parameters
+        self.calculations = calculations
+        self.visibilities = visibilities
+
+    @classmethod
+    def new(cls, safe) -> LuxtronikData:
+        return LuxtronikData(Parameters(safe), Calculations(), Visibilities())
+
+    @classmethod
+    def copy(cls, data) -> LuxtronikData:
+        return LuxtronikData(copy(data.parameters), copy(data.calculations), copy(data.visibilities))
+
+
+class LuxtronikInterface:
```

Comment:
`LuxtronikSocketInterface` would be more descriptive?

Reply:
Done